WickedEngine/wiMath.h

```diff
@@ -17,66 +17,111 @@ namespace wiMath
 	}
 	inline float Distance(const XMVECTOR& v1, const XMVECTOR& v2)
 	{
+#ifdef _WIN32
 		XMVECTOR& vectorSub = XMVectorSubtract(v1, v2);
 		XMVECTOR& length = XMVector3Length(vectorSub);
+#else
+		XMVECTOR vectorSub = DirectX::XMVectorSubtract(v1, v2);
+		XMVECTOR length = DirectX::XMVector3Length(vectorSub);
+#endif
 
 		float Distance = 0.0f;
 		XMStoreFloat(&Distance, length);
 		return Distance;
 	}
 	inline float DistanceSquared(const XMVECTOR& v1, const XMVECTOR& v2)
 	{
+#ifdef _WIN32
 		XMVECTOR& vectorSub = XMVectorSubtract(v1, v2);
 		XMVECTOR& length = XMVector3LengthSq(vectorSub);
+#else
+		XMVECTOR vectorSub = DirectX::XMVectorSubtract(v1, v2);
+		XMVECTOR length = DirectX::XMVector3LengthSq(vectorSub);
+#endif
 
 		float Distance = 0.0f;
 		XMStoreFloat(&Distance, length);
 		return Distance;
 	}
 	inline float DistanceEstimated(const XMVECTOR& v1, const XMVECTOR& v2)
 	{
+#ifdef _WIN32
 		XMVECTOR& vectorSub = XMVectorSubtract(v1, v2);
 		XMVECTOR& length = XMVector3LengthEst(vectorSub);
+#else
```

Comment:
Can we just be using the DirectX namespace globally as it is right now? It will be a pain in the ass to ifdef every single math usage per platform.

Reply:
Yes. That was just to get my ide to stop complaining, so I could find actual problems easier.